ONSdigital/sdc-int-rh-ui
# Catch Redis connection and unknown http errors

# Motivation and Context

Currently the RH UI presents the user with a python traceback if redis isn't running. We want to present them with a formatted error page.
# What has changed



Error handler middleware inserted into the list of middlewares so it sets itself up first. Redis connection errors caught and unknown aiohttp errors also caught.
# How to test?



Build the RH-UI off this branch and do a make up.
Open RH-UI start page in a browser and check it displays correctly, and that it creates a cookie in your browser called `RH_SESSION`
Run `docker stop redis`
Open RH-UI in your browser again and check it displays a formatted HTML page saying 'Sorry, something went wrong`
Delete the `RH_SESSION` cookie and refresh the page, check the same error page displays correctly

Do the same as above but for cy and check the pages are displayed in Welsh

run a `make up` in docker dev to bring redis back up and run the ATs
# Links



https://trello.com/c/HzqUNDxz/555-rh-error-page-on-redis-failure
# Screenshots (if appropriate):
